### DISCUSSION TITLE
How can I highlight React Code in VSCode?

### DISCUSSION THREAD

**Discussion by mohdh34m** [microsoft/vscode-discussions#100] [Extension Development QnA] [answered]
**I'm developing my own VSCode Theme But I have a problem that the HTML codes in react appear in one color**

The picture below shows the problem:

![](https://i.stack.imgur.com/N7PL7.png)

**Comment by gjsjohnmurray** [comment, marked as answer]
I suggest you use the advice [here](https://code.visualstudio.com/docs/getstarted/themes#_editor-semantic-highlighting), particularly about using the scope inspector (Developer: Inspect Editor Tokens and Scopes) to discover how your React code is being marked up by whichever extension you are using for it.

**Reply by mohdh34m** [reply]
TextMate inspector shows me "no theme selector"

[![Screenshot-2022-08-14-175335.png](https://i.postimg.cc/BnLS5RFX/Screenshot-2022-08-14-175335.png)](https://postimg.cc/hXnRcypB)
